src/main/java/com/apifest/doclet/Doclet.java

```diff
@@ -82,188 +104,333 @@
-            return false;
-        }
-
-        backendHost = System.getProperty("backend.host");
-        if(backendHost == null || backendHost.length() == 0 || NULL.equals(backendHost)) {
-            System.out.println("ERROR: backend.host is not set");
-            return false;
-        }
-
-        String backendPortStr = System.getProperty("backend.port");
-        if(backendPortStr == null || backendPortStr.length() == 0 || NULL.equals(backendPort)) {
-            System.out.println("ERROR: backend.port is not set");
-            return false;
-        }
-
         try {
-            backendPort = Integer.valueOf(backendPortStr);
-        } catch (NumberFormatException e) {
-            System.out.println("ERROR: backendPort is not an integer");
+            validateConfiguration();
+        } catch (IllegalArgumentException ex) {
+            System.out.println("ERROR: " + ex.getMessage());
             return false;
         }
 
-        defaultActionClass = System.getProperty("defaultActionClass");
-
-        defaultFilterClass = System.getProperty("defaultFilterClass");
-
-        outputFile = System.getProperty("mapping.filename");
-
-        applicationPath = System.getProperty("application.path");
-
         System.out.println("Start ApiFest Doclet>>>>>>>>>>>>>>>>>>>");
         System.out.println("mapping.version is: " + System.getProperty("mapping.version"));
-        System.out.println("backend.host: " + System.getProperty("backend.host"));
-        System.out.println("backend.port: " + System.getProperty("backend.port"));
 
         try {
+            List<ParsedEndpoint> parsedEndpoints = new ArrayList<ParsedEndpoint>();
             ClassDoc[] classes = root.classes();
             for (ClassDoc clazz : classes) {
                 MethodDoc[] mDocs = clazz.methods();
                 for (MethodDoc doc : mDocs) {
-                    MappingEndpoint endpoint = getMappingEndpoint(doc);
-                    if(endpoint != null) {
-                        endpoints.add(endpoint);
+                    ParsedEndpoint parsed = parseEndpoint(doc);
+                    if (parsed != null) {
+                        parsedEndpoints.add(parsed);
                     }
                 }
             }
-
-            generateMappingFile(outputFile);
-        } catch (JAXBException e) {
+            if (docletMode.contains(DocletMode.DOC)) {
+                generateDocsFile(parsedEndpoints, mappingDocsOutputFile);
+            }
+            if (docletMode.contains(DocletMode.MAPPING)) {
+                generateMappingFile(parsedEndpoints, mappingOutputFile);
+            }
+            return true;
+        } catch (ParseException e) {
             System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
             return false;
-        } catch (ParseException e) {
+        } catch (JsonGenerationException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
```

Comment:
Here the error message should contain "cannot create mapping docs file" instead of "mapping file". The same is true for the next error messages as well.